be1-go/hub/standard_hub/mod.go

```diff
@@ -495,12 +463,8 @@ func (h *Hub) handleIncomingMessage(incomingMessage *socket.IncomingMessage) err
 
 // sendGetMessagesByIdToServer sends a getMessagesById message to a server
 func (h *Hub) sendGetMessagesByIdToServer(socket socket.Socket, missingIds map[string][]string) error {
-	h.Lock()
-	defer h.Unlock()
-
-	queryId := h.queries.nextID
-	baseValue := false
-	h.queries.state[queryId] = &baseValue
+	queryId := h.queries.GetNextID()
+	h.queries.SetQueryState(queryId, false)
```

Comment:
shouldn't this function be executed _after_ the `json.Marshal` succeeds?  (and right before the `AddQuery` ?
Also, should the query state be set automatically if/when we do an `AddQuery` ?

Reply:
Yes, good catch. I'll change that and also make the state set automatic and leave the function for changing the state only upon reception.